In [1]:
import pandas as pd

df = pd.read_csv('../datasets/reviews.csv')

In [2]:
df.head()

id       asins   brand                  categories  \
0  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   
1  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   
2  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   
3  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   
4  AVpe7AsMilAPnD_xQ78G  B00QJDU3KY  Amazon  Amazon Devices,mazon.co.uk   

  colors             dateAdded           dateUpdated  \
0    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   
1    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   
2    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   
3    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   
4    NaN  2016-03-08T20:21:53Z  2017-07-18T23:52:58Z   

                  dimension  ean                         keys  ...  \
0  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   
1  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   
2  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   
3  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   
4  169 mm x 117 mm x 9.1 mm  NaN  kindlepaperwhite/b00qjdu3ky  ...   

  reviews.rating                                 reviews.sourceURLs  \
0            5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
1            5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
2            4.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
3            5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   
4            5.0  https://www.amazon.com/Kindle-Paperwhite-High-...   

                                        reviews.text  \
0  I initially had trouble deciding between the p...   
1  Allow me to preface this with a little history...   
2  I am enjoying it so far. Great for reading. Ha...   
3  I bought one of the first Paperwhites and have...   
4  I have to say upfront - I don't like coroporat...   

                                reviews.title reviews.userCity  \
0              Paperwhite voyage, no regrets!              NaN   
1           One Simply Could Not Ask For More              NaN   
2  Great for those that just want an e-reader              NaN   
3                    Love / Hate relationship              NaN   
4                                   I LOVE IT              NaN   

  reviews.userProvince    reviews.username  sizes upc     weight  
0                  NaN          Cristina M    NaN NaN  205 grams  
1                  NaN               Ricky    NaN NaN  205 grams  
2                  NaN       Tedd Gardiner    NaN NaN  205 grams  
3                  NaN              Dougal    NaN NaN  205 grams  
4                  NaN  Miljan David Tanic    NaN NaN  205 grams  

[5 rows x 27 columns]

In [3]:
print('Number of reviews: ', len(df))

Number of reviews:  1597


`reviews.csv` counts with 1597 review samples. Some of the columns are not relevant for our model trainning, relevant columns are:
- `reviews.rating`
- `reviews.title`
- `reviews.text`

Other columns can be dropped.

In [4]:
df = df.filter(['reviews.rating', 'reviews.title', 'reviews.text'], axis=1)
df.head(10)

reviews.rating                                      reviews.title  \
0             5.0                     Paperwhite voyage, no regrets!   
1             5.0                  One Simply Could Not Ask For More   
2             4.0         Great for those that just want an e-reader   
3             5.0                           Love / Hate relationship   
4             5.0                                          I LOVE IT   
5             NaN  Great device for reading. 8 people found this ...   
6             NaN  One Simply Could Not Ask For More 28 people fo...   
7             NaN  Definitely better than the previous generation...   
8             NaN  Paperwhite voyage, no regrets! 16 people found...   
9             NaN  Great for those that just want an e-reader 19 ...   

                                        reviews.text  
0  I initially had trouble deciding between the p...  
1  Allow me to preface this with a little history...  
2  I am enjoying it so far. Great for reading. Ha...  
3  I bought one of the first Paperwhites and have...  
4  I have to say upfront - I don't like coroporat...  
5  My previous kindle was a DX, this is my second...  
6  Allow me to preface this with a little history...  
7  Just got mine right now. Looks the same as the...  
8  I initially had trouble deciding between the p...  
9  I am enjoying it so far. Great for reading. Ha...

In [5]:
# rename collums to remove dots
df = df.rename({ 'reviews.rating': 'rating', 'reviews.title': 'title', 'reviews.text': 'text' }, axis=1)

In [6]:
df.head(10)

rating                                              title  \
0     5.0                     Paperwhite voyage, no regrets!   
1     5.0                  One Simply Could Not Ask For More   
2     4.0         Great for those that just want an e-reader   
3     5.0                           Love / Hate relationship   
4     5.0                                          I LOVE IT   
5     NaN  Great device for reading. 8 people found this ...   
6     NaN  One Simply Could Not Ask For More 28 people fo...   
7     NaN  Definitely better than the previous generation...   
8     NaN  Paperwhite voyage, no regrets! 16 people found...   
9     NaN  Great for those that just want an e-reader 19 ...   

                                                text  
0  I initially had trouble deciding between the p...  
1  Allow me to preface this with a little history...  
2  I am enjoying it so far. Great for reading. Ha...  
3  I bought one of the first Paperwhites and have...  
4  I have to say upfront - I don't like coroporat...  
5  My previous kindle was a DX, this is my second...  
6  Allow me to preface this with a little history...  
7  Just got mine right now. Looks the same as the...  
8  I initially had trouble deciding between the p...  
9  I am enjoying it so far. Great for reading. Ha...